app/models/company.rb

```diff
@@ -117,6 +117,34 @@ def self.next_branding_payment_dates(company_id)
     next_payment_dates(company_id, Payment::PAYMENT_TYPE_BRANDING)
   end
 
+  # The first day the H-branding fee is past due (a.k.a "day zero") is
+  #    IF an H-branding fee has ever been paid,
+  #      it is 1 day after the latest expiration date of the latest H-branding fee payment
+  #    ELSE
+  #      it is the date of the _earliest_ membership fee payment of the current members in the company.
+  #
+  #
+  # If a company has previous H-Brand payments and the latest *expiration date is 30 Nov 2017*
+  #    then the day that the next H-Brand payment is due is *1 Dec 2017*
+  #
+  # That means that the last H-Brand payment was made 1 Dec 2016.
+  #  The H-Brand licensing term ran from 1 Dec 2016 to 30 Nov 2017.
+  #  Thus the next H-Brand payment is due 1 Dec 2017.
+  #
+  # On 1 Dec 2017, the H-Brand fee would be zero (0) days _past due_.
+  # On 2 Dec 2017, the H-Brand fee would be 1 days _past due._
+  # Thus the `0 days past due` date is 1 Dec 2017.
+  #
+  #
+  # @return [Date]  - the Date that is the day that the branding payment becomes past due.
```

Comment:
This logic is different from the other alerts logic.  Here, we're using a concept of 'past due' as day zero.  Elsewhere, we're using the concept of 'expiration day' as day zero.

*We need to standardize on one way to do this.*

One reason is so email text contains the right language.  Such as " ... your next payment was due 7 days ago ..." as opposed to " ... you prior payment expired 8 days ago ...".

I also have some discomfort with the "past due" language.  Is it past due?  I don't think so.  It is "past due" if we're providing a continuing service through and past the prior expire date - but we're not.  We're actually turning off visibility, preventing access to branding credential etc.  

This is why I much prefer to talk about "days since your prior payment expired".  Language referring to "past due" strikes me as possibly being received as accusatory, and might antagonize some people.

Reply:
I totally agree we need to standardize on the definitions used in the code _and_ which methods are used where, and why.  
_= the models_

But let's be sure to separate that from what we need to display to users.  (views, mailer views, etc.)  We can present things to users in whatever manner is appropriate, using whatever wording or display is best, whether that's using "... expired" or "... past due'"

The underlying models  always calculate the value correctly no matter which is requested by a view.

For this particular feature, we are (and have been) specifically talking about days _past due_.  So the model needs to be able to provide that information. 

We definitely need to be sure that we are consistent with what is in the Condition configuration information (e.g. days: [.....]).  And that the usage is clear and well documented.

Reply:
@patmbolger After our meeting this morning and looking at this again, I have a question about some terminology for this feature.

I started thinking about this because I thought I should change the name of the alert class from "...PastDue.."  to something else. 

There are 2 different possible situations we need to cover for the company:
1.  There has been an H-Branding license term in the past.  Which means there is a payment with an expiration date.   The terminology is straightforward:  "The previous term _expired_ on <whatever date>"  

2. There has never been an H-Branding license term for the company in the past.  There has never been an H-branding license payment made.   In this situation, we decided that the right business logic is to use the date for the _earliest current member fee paid_.  There is no "expired" date that comes in to play here.

Seems that  the answer is that we just call this alert something like "HBrandingFee**Due**" ?
(no "past" or "expired" word in there)

Thoughts?

(Also -- I know that if we change the name of the `HBrandingFeePastDue` conditionResponder (alert) class, we'll need to be sure to change it in the Conditions table in the production db.  This would be done via a rake task that would need to also be included in the PR.)